__tests__/__integration__/imperative/__tests__/__integration__/cli/config/list/cli.imperative-test-cli.config.list.integration.test.ts

```diff
@@ -0,0 +1,185 @@
+        expect(response.stdout.toString()).toMatchSnapshot();
+        expect(response.stdout.toString()).toContain("secured: my_secured");
+        expect(response.stdout.toString()).toContain("type:       secured");
+        expect(response.stdout.toString()).toContain("defaults:");
+        expect(response.stdout.toString()).toContain("profiles:");
+        expect(response.stdout.toString()).toContain("plugins:");
+        expect(response.stdout.toString()).toContain("secure:");
+        expect(response.stdout.toString()).toContain("(empty array)");
+        expect(response.stderr.toString()).toEqual("");
+        expect(response.error).not.toBeDefined();
+    });
+    it("should list the configuration in RFJ", () => {
+        const response = runCliScript(__dirname + "/__scripts__/list_config.sh", TEST_ENVIRONMENT.workingDir, ["--rfj"]);
+        const parsedResponse = JSON.parse(response.stdout.toString());
+        const expectedResponse = {
+            data: {
+                defaults: {
+                    secured: "my_secured"
+                },
+                profiles: {
+                    my_secured: {
+                        type: "secured",
+                        properties: {
+                            info: "",
+                        }
+                    }
+                },
+                plugins: [],
+                secure: []
+            }
+        };
+        expect(parsedResponse.success).toEqual(true);
+        expect(parsedResponse.stderr).toEqual("");
+        expect(parsedResponse.exitCode).toEqual(0);
+        expect(parsedResponse.data).toEqual(expectedResponse.data);
+    });
+    it("should list the configurations based on location", () => {
+        const response = runCliScript(__dirname + "/__scripts__/list_config.sh", TEST_ENVIRONMENT.workingDir, ["--locations"]);
+        expect(response.stdout.toString()).toContain(expectedProjectConfigLocation);
+        expect(response.stdout.toString()).toContain(expectedUserConfigLocation);
+        expect(response.stdout.toString()).toContain(expectedGlobalProjectConfigLocation);
+        expect(response.stdout.toString()).toContain(expectedGlobalUserConfigLocation);
+        expect(response.stdout.toString()).toContain("defaults:");
+        expect(response.stdout.toString()).toContain("profiles:");
+        expect(response.stdout.toString()).toContain("plugins:");
+        expect(response.stdout.toString()).toContain("secure:");
+        expect(response.stdout.toString()).toContain("my_secured:");
+        expect(response.stdout.toString()).toContain("type:       secured");
+        expect(response.stdout.toString()).toContain("properties:");
+        expect(response.stdout.toString()).toContain("secured: my_secured");
+        expect(response.stdout.toString()).toContain("$schema:  ./schema.json");
+        expect(response.stderr.toString()).toEqual("");
+        expect(response.error).not.toBeDefined();
```

Comment:
Thanks for not using snapshots. ❤️ 

Curious where you think we should draw the line (for when to use them and when not to use them).

No changes requested!.

Reply:
This one, we didn't want to use snapshots because there are file paths in the object, so using snapshots would cause false positives to arise here. I agree that there are a lot of things being looked for here, though.